src/test/java/com/cloudant/tests/ViewsTest.java

```diff
@@ -267,6 +481,15 @@ public void paginationAscendingReusingViewAllTheWayInEachDirectionTwice() {
     }
 
     /**
+     * Run the same check as paginationAscendingReusingViewAllTheWayInEachDirectionTwice
+     * test on views using an array of multi-value keys.
+     */
+    @Test
+    public void paginationAscendingReusingMVKeyViewAllTheWayInEachDirectionTwice() {
+        checkPaginationWithMultiValueKey(false, 6, 2, 3, 1, 3, 1);
```

Comment:
You could use a simple builder approach using inner classes to make these tests much more readable.

```java
public class CheckPaginationWithMultiValueKey {
 public b() {
   return new CheckPaginationWithMultiValueKey();
  }
  public descending(bool descending) {
    this.descending = descending;
    return this;
  }
  /* etc for other args */
  public runTest() {
    /* do current content of checkPaginationWithMultiValueKey method here */
  }
}